go/common/viewingkey/viewing_key.go

```diff
@@ -103,9 +106,48 @@ func GenerateSignMessage(vkPubKey []byte) string {
 	return SignedMsgPrefix + hex.EncodeToString(vkPubKey)
 }
 
-// GenerateSignMessageOG creates the message to be signed by Obscuro Gateway (new format)
-// format is expected to be "Register <userID> for <Account>" (with the account in lowercase)
-func GenerateSignMessageOG(vkPubKey []byte, addr *gethcommon.Address) string {
-	userID := crypto.Keccak256Hash(vkPubKey).Bytes()
-	return fmt.Sprintf("Register %s for %s", hex.EncodeToString(userID), strings.ToLower(addr.Hex()))
+// GetEIP712RawData Get raw data (bytes) for an EIP-712 message used to authenticate an address with user
+func GetEIP712RawData(userID string, address *gethcommon.Address) ([]byte, error) {
+	types := apitypes.Types{
+		"EIP712Domain": {
+			{Name: "name", Type: "string"},
+			{Name: "version", Type: "string"},
+			{Name: "chainId", Type: "uint256"},
+			{Name: "verifyingContract", Type: "address"},
+		},
+		"Register": {
+			{Name: "userID", Type: "string"},
+			{Name: "account", Type: "address"},
+		},
+	}
+
+	domain := apitypes.TypedDataDomain{
+		Name:              "Obscuro",
+		Version:           "1.0",
+		ChainId:           (*math.HexOrDecimal256)(big.NewInt(443)),
```

Comment:
we should save the chainId that as a variable.
Doesn't it have to be configurable? or read from the backend node? To work on testnets, mainnets, etc